common/djangoapps/enrollment/views.py

```diff
@@ -342,23 +341,16 @@ def post(self, request):
         """
         Unenrolls the specified user from all courses.
         """
-        username = None
-        user_model = get_user_model()
-
         try:
-            # Get the username from the request and check that it exists
+            # Get the username from the request.
             username = request.data['username']
```

Comment:
I've gone back and forth on this some, but if this is a retirement-only endpoint (as the permission required to hit it is `CanRetireUser` can we change this to use `get_retirement_for_retirement_action` and bring in those safeguards? We do that in similar places and 404 if that entry isn't found.

Reply:
Ah - right. I was thinking about this being a general-purpose endpoint when I made the change - but you're correct that this is actually a retirement-only endpoint as it stands right now.

I'll change this.